src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/HealthCheckSpecification.groovy

```diff
@@ -22,100 +23,238 @@ class HealthCheckSpecification extends HealthCheckBaseSpecification {
-            server42FlowRtt
-            server42IslRtt
-            //for below props any value allowed. dependent tests will skip themselves or adjust if feature is off
-            //flowsRerouteUsingDefaultEncapType
-            //collectGrpcStats
+
+        Closure allSwitchesConnectedToExpectedRegion = {
+            def regionVerifications = new SoftAssertions()
+            flHelper.fls.forEach { fl ->
+                def expectedSwitchIds = topology.activeSwitches.findAll { fl.region in it.regions }*.dpId
+                if (!expectedSwitchIds.empty) {
+                    regionVerifications.checkSucceeds {
+                        assert fl.floodlightService.switches*.switchId.sort() == expectedSwitchIds.sort()
+                    }
+                }
+            }
+            regionVerifications.verify()
         }
 
-        // and: "Switches configurations are in expected state"
-        topology.activeSwitches.each { sw ->
-            verifyAll(northbound.getSwitchProperties(sw.dpId)) {
-                multiTable == useMultitable && sw.features.contains(SwitchFeature.MULTI_TABLE)
-                //server42 props can be either on or off
+        Closure featureTogglesInExpectedState = {
+            verifyAll(northbound.getFeatureToggles()) {
+                flowsRerouteOnIslDiscoveryEnabled
+                createFlowEnabled
+                updateFlowEnabled
+                deleteFlowEnabled
+                useBfdForIslIntegrityCheck
+                floodlightRoutePeriodicSync
+                server42FlowRtt
+                server42IslRtt
+                //for below props any value allowed. dependent tests will skip themselves or adjust if feature is off
+                //flowsRerouteUsingDefaultEncapType
+                //collectGrpcStats
+            }
+        }
+        Closure switchesConfigurationIsCorrect = {
+            withPool {
+                topology.activeSwitches.eachParallel { sw ->
+                    verifyAll(northbound.getSwitchProperties(sw.dpId)) {
+                        multiTable == useMultitable && sw.features.contains(SwitchFeature.MULTI_TABLE)
+                        //server42 props can be either on or off
+                    }
+                }
             }
         }
+        withPool {
+            [healthCheckEndpoint,
+             allZookeeperNodesInExpectedState,
+             allSwitchesAreActive,
+             allLinksAreActive,
+             noFlowsLeft,
+             noLinkPropertiesLeft,
+             linksBandwidthAndSpeedMatch,
+             noExcessRulesMeters,
+             allSwitchesConnectedToExpectedRegion,
+             featureTogglesInExpectedState,
+             switchesConfigurationIsCorrect].eachParallel {
+                println("Starting method ${it.getClass().getSimpleName()} at ${new Date()}")
+                it()
+                println("Ending method ${it.getClass().getSimpleName()} at ${new Date()}")
+            }
+        }
```

Comment:
jfyi, you can also put it inside `Wrappers.benchmark` method for debug purposes to check the performance gain compared to how it was before

Reply:
sorry for reviewing 😁